force-app/main/default/classes/ADDR_Addresses_UTIL_TEST.cls

```diff
@@ -0,0 +1,1341 @@
+
+        Test.startTest();
+        Boolean isAddressEmpty = ADDR_Addresses_UTIL.isAddressEmpty(addresses[0]);
+        Test.stopTest();
+
+        System.assertEquals(false, isAddressEmpty);
+    }
+
+    @isTest
+    static void testIsAddressEmptyTrue() {
+        //This will fail every time we adding a new field
+        List<Address__c> addresses = UTIL_UnitTestData_TEST.getMultipleTestAddresses(1);
+        addresses[0].MailingStreet__c = null;
+        addresses[0].MailingStreet2__c = null;
+        addresses[0].MailingCity__c = null;
+        addresses[0].MailingCounty__c = null;
+        addresses[0].MailingState__c = null;
+        addresses[0].MailingPostalCode__c = null;
+        addresses[0].MailingCountry__c = null;
+
+        Test.startTest();
+        Boolean isAddressEmpty = ADDR_Addresses_UTIL.isAddressEmpty(addresses[0]);
+        Test.stopTest();
+
+        System.assertEquals(true, isAddressEmpty);
+    }
+
+    @isTest
+    static void testIsAddressEmptyNull() {
+        //This is working but null address is not empty does not make sense
+        Test.startTest();
+        Boolean isAddressEmpty = ADDR_Addresses_UTIL.isAddressEmpty(null);
+        Test.stopTest();
+
+        System.assertEquals(false, isAddressEmpty);
+    }
+
+    /**************************************************************************************************************************
+     ****************************************************** isContactAccAddrInfoSame ******************************************
+     **************************************************************************************************************************/
+    @isTest
+    static void isContactAccAddrInfoSameTrue() {
+        Contact contact = UTIL_UnitTestData_TEST.getContact();
+        List<Account> accounts = UTIL_UnitTestData_TEST.getMultipleTestAccounts(
+            1,
+            UTIL_Describe_API.getBizAccRecTypeID()
+        );
+
+        Test.startTest();
+        Boolean isContactAccAddrInfoSame = ADDR_Addresses_UTIL.isContactAccAddrInfoSame(contact, accounts[0]);
+        Test.stopTest();
+
+        System.assertEquals(true, isContactAccAddrInfoSame);
+    }
+
+    @isTest
+    static void isContactAccAddrInfoSameFalseDifferentAddressField() {
+        Contact contact = UTIL_UnitTestData_TEST.getContact();
+        List<Account> accounts = UTIL_UnitTestData_TEST.getMultipleTestAccounts(
+            1,
+            UTIL_Describe_API.getBizAccRecTypeID()
+        );
+        contact.MailingCity = 'Something else';
+
+        Test.startTest();
+        Boolean isContactAccAddrInfoSame = ADDR_Addresses_UTIL.isContactAccAddrInfoSame(contact, accounts[0]);
+        Test.stopTest();
+
+        System.assertEquals(false, isContactAccAddrInfoSame);
+    }
+
+    @isTest
+    static void isContactAccAddrInfoSameFalseDifferentStateCode() {
+        /*Almost no way we can test state code/country code here
```

Comment:
Given this use case, make a note for TODO at the appropriate locations in ADDR_Addresses_UTIL so we surface each of these use cases for posterity.

Reply:
I came with the idea that we could add refactoring comments to test class:  1) Hoping that whenever we are refactoring any function, we go to the related unit tests first 2) We don't add TODO comments to our functional code so that it is nice and clean.

Reply:
I'm comfortable with that approach.